treeherder/perfalert/perfalert/__init__.py

```diff
@@ -89,77 +89,69 @@ def __repr__(self):
                                                  self.t, self.state)
 
 
-class Analyzer:
-    def __init__(self):
-        self.data = []  # List of PerfDatum instances
-
-    def add_data(self, push_timestamp, value, **kwargs):
-        self.data.append(Datum(push_timestamp, value, **kwargs))
-        self.data.sort()
-
-    def analyze_t(self, min_back_window=12, max_back_window=24,
-                  fore_window=12, t_threshold=7):
-        # Use T-Tests
-        # Analyze test data using T-Tests, comparing data[i-j:i] to data[i:i+k]
-        (j, k) = (min_back_window, fore_window)
-        good_data = []
-
-        num_points = len(self.data) - k + 1
-        last_seen_regression = 0
-        for i in range(num_points):
-            di = self.data[i]
-            # if we haven't seen something that looks like a change in a
-            # while, incorporate some extra historical data into our t-test
-            # calculation
-            if last_seen_regression > min_back_window:
-                additional_back_window = min(last_seen_regression,
-                                             max_back_window - min_back_window)
+def detect_changes(data=[], min_back_window=12, max_back_window=24, 
```

Comment:
data shouldn't be a kwargs, as it doesn't make sense to call this function without it. just write this as detect_changes(data, ...)